x-pack/plugins/stack_connectors/common/slack/schema.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+export const SlackConfigSchema = schema.object({
+  type: schema.oneOf([schema.literal('webhook'), schema.literal('web_api')]),
+});
+
+export const SlackWebhookSecretsSchema = schema.object({
+  webhookUrl: schema.string({ minLength: 1 }),
+});
+export const SlackWebApiSecretsSchema = schema.object({
+  token: schema.string({ minLength: 1 }),
+});
+
+export const SlackSecretsSchema = schema.oneOf([
```

Comment:
Hi, @pmuellr !
It's not possible to update existing connector from UI. I made it intentional, because for Web API connector you still need to go every rule and choose a slack channel. And it will be a lot of errors before you choose one. I did not test this behaviour, because I decided if every rule will still need changes, it make sense to make connector type is not changeable.
I did not think about that customer can change connector type just by calling API. Not sure how to proceed with that. Make it somehow impossible?
What do you think?

Reply:
Yeah, users won't be able to just "switch" from old to new, without picking a channel.  But that's ok, we already have situations like this.  For instance, you can change email from authentication not required, to required, which will force you to enter in a userid / password.

On the other hand, maybe it's not a big deal.  Not sure how many users might want to "upgrade" an existing Slack connector to the new capabilities.  The big win would be not having to change connectors already used in alerting actions.  @shanisagiv1 ?

Reply:
I agree it might be annoying to switch from old to new in the existing connector and then all your slack actions will be failed until you actively fix that. (it's not exactly the same as forcing me to type in cred when email auth is changed since you don't need to go 1 by 1 and change all your email actions.. )

I'd prefer for new slack connectors to open the new tab by default so users will use the new functionality. 
And for users that want to "upgrade" - They'll have to create a new connector and anyway they will have to change (replace) the actions within the relevant rules. 
So maybe for existing slack instances, we need to hide the new tab if it makes sense.
Sorry if it's not aligned with what you already did. Can I test it out somewhere?

wdyt? Thanks

Reply:
@shanisagiv1 It's exactly how it works now. 
But some improvement can be done I think. Like if we see that customer wants to edit Slack connector and change type to Web API, we can somehow show module with channel picker or something like this. 

Just want understand do we want to do it and what's the precedence.

Reply:
About test it. I think I heard about tag, which will create during the build a running in the cloud kibana instance based on current branch. I'll ask around..

Reply:
I say we leave it as is and don't allow a user to change, right now.  If there's interest, we could add it later, but likely it will have less value later, as most users will be using the new one anyway.

Reply:
I added a `ci:cloud-deploy` which should push a cloud deployment.  I mistakenly first added `ci:cloud-redeploy` (which probably removes an existing deployment and rebuilds), but then changed it back to `ci:cloud-deploy`.  Maybe that'll get screwed up, but then we can likely remove the label, and then add it back a bit later (or use `ci:cloud-redeploy`.